woocommerce/woocommerce-ios
# Native Jetpack Installation: Apply A/B test

Closes: #8166 

### Description
This PR adds a new A/B test experiment for simplified login to enable native Jetpack installation flow. Experiment details: 

- Added new A/B test experiment.
- Replaced the temporary feature flag with A/B test.

### Testing instructions

Prerequisite: Make sure that you have a test store that is not associated with your WP.com account or doesn't have Jetpack.

#### `control` variation

- Log out of the app or skip login onboarding if needed.
- On the prologue screen, select "Log In" or "Continue with WordPress.com" based on the A/B test variant you get.
- Log in with your WordPress.com account.
- If your account doesn't have any associated site, notice that the empty store picker is displayed with the matching design as on Figma.
- Select Add a Store > Connect an existing site.
- Enter the address of your test store and tap Continue.
- You will see the existing Jetpack error screen. 
- You can install and setup Jetpack in a webview from this screen.

#### Testing the `treatment` variation

We will have to sandbox and assign a particular variation to the device to test the variations.

Prerequisites:
- Please follow the setup in PCYsg-Fq7-p2#assignments-api for proxied sandboxed API requests: connect to sandbox via ssh, and point WP.com API to the sandbox IP address (`ifconfig`) in the local matchine's `/etc/hosts`
- In Xcode, set a breakpoint at `ExPlatService` L47 where `anon_id` is set in the URL request
- Make an API request `PATCH /wpcom/v2/experiments/0.1.0/assignments` with the following `body` and make sure it's successful:
```
{
    "variations": {
        "woocommerceios_login_jetpack_setup_flow": "treatment"
    },
    "anon_id": "anon_id_from_the_previous_step",
    "username_override": ""
}
```
- Launch the app, and let it sit for a bit
- Close and relaunch the app 
- Log out of the app or skip login onboarding if needed.
- On the prologue screen, select "Log In" or "Continue with WordPress.com" based on the A/B test variant you get.
- Log in with your WordPress.com account.
- If your account doesn't have any associated site, notice that the empty store picker is displayed with the matching design as on Figma.
- Select Add a Store > Connect an existing site.
- Enter the address of your test store and tap Continue.
- You will see the new Jetpack error screen. (Screenshot below)
- You can install and setup Jetpack in new native flow starting from this screen.

### Screenshots

| Control | Treatment |
| ----- | ----- |
| <img src="https://user-images.githubusercontent.com/524475/203035274-3f7d0caf-76a0-44a8-ad6c-fb7f0b43b489.png" width=320 /> | <img src="https://user-images.githubusercontent.com/524475/203014719-3b2fc035-1aff-460a-b49c-94b86e9bd846.png" width=320 /> | 



---
